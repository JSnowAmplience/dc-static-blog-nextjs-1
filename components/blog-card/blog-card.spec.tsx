/* eslint-env jest */
import renderer from 'react-test-renderer';
import BlogCard from './blog-card';
import blogPostFixture from '../../tests/fixtures/single-blog-post-data-object.json';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import StaticLink from '../static-link/static-link';

const mockUseRouter = jest.fn();
jest.mock('next/router', () => {
  return {
    useRouter: () => mockUseRouter()
  };
});

describe('BlogCard', () => {
  test('renders full blog card', () => {
    mockUseRouter.mockImplementationOnce(() => {
      return { query: {} };
    });
    const component = mount(<BlogCard blogPost={blogPostFixture} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  test('renders full blog card with lazy loaded placeholders', () => {
    mockUseRouter.mockImplementationOnce(() => {
      return { query: {} };
    });
    const component = renderer.create(<BlogCard blogPost={blogPostFixture} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders full blog card with vse query string in link for preview', () => {
    mockUseRouter.mockImplementationOnce(() => {
      return {
        query: { vse: 'test-vse.domain' }
      };
    });

    const component = mount(<BlogCard blogPost={blogPostFixture} />);
    expect(component.find('a').props().href).toEqual(
      '/preview?vse=test-vse.domain&content=8d6943c7-6028-4fac-b45e-57fc63bd032a'
    );
  });
});
