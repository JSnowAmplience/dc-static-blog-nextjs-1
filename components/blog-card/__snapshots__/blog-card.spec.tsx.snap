// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard renders full blog card 1`] = `
<section
  className="jsx-3182958052"
>
  <a
    aria-label="blog-post-test"
    className="jsx-3354872520"
    href="/blog/url-slug"
  >
    <article
      className="jsx-3182958052"
    >
      <div
        className="jsx-3182958052 blog-card-image"
      >
        <div
          className="image-mock"
        />
      </div>
      <div
        className="jsx-3182958052 blog-card-content"
      >
        <h1
          className="jsx-3182958052"
        >
          blog-post-test
        </h1>
        <div
          className="blog-card-meta-mock"
        />
        <p
          className="jsx-3182958052"
        >
          blog-post-description
        </p>
      </div>
    </article>
  </a>
</section>
`;
